packages/search/src/style.scss

```diff
@@ -0,0 +1,253 @@
+		flex: 0 0 auto;
+		width: 50px;
+		z-index: 20;
+		color: var( --color-primary );
+		cursor: pointer;
+	}
+
+	.search__open-icon:hover {
+		color: var( --color-neutral-60 );
+	}
+
+	.search__close-icon {
+		color: var( --color-neutral-60 );
+		opacity: 0;
+		transition: opacity 0.2s ease-in;
+	}
+
+	&.is-open.has-focus {
+		border-color: var( --color-primary );
+		box-shadow: 0 0 0 1px var( --color-primary ), 0 0 0 4px var( --color-primary-10 );
+
+		&:hover {
+			box-shadow: 0 0 0 1px var( --color-primary ), 0 0 0 4px var( --color-primary-20 );
+		}
+	}
+
+	&.is-compact,
+	&.is-expanded-to-container.is-compact {
+		height: 36px;
+		justify-content: end;
+		.search__input {
+			display: flex;
+			align-items: center;
+			height: 36px;
+			font-size: $font-body-extra-small;
+		}
+
+		.search__open-icon {
+			margin: 0 4px 0 8px;
+			width: 18px;
+		}
+
+		.search__close-icon {
+			width: 18px;
+			margin-right: 8px;
+		}
+	}
+}
+
+// Position collapsed search-button to the right
+// of the container element
+.search.is-expanded-to-container {
+	margin-bottom: 0;
+	position: absolute;
+	display: flex;
+	height: 100%;
+	width: 50px;
+	top: 0;
+	right: 0;
+
+	.search__input-fade {
+		position: relative;
+		flex: 0 0 auto;
+		display: flex;
+		align-items: center;
+	}
+
+	.search__input {
+		flex: 1 1 auto;
+		display: flex;
+	}
+}
+
+.search:not( .is-open ) input.search__input[type='search'] {
+	display: none;
+}
+
+.search input.search__input[type='search'] {
+	flex: 1 1 auto;
+	z-index: $input-z-index;
+	top: 0;
+	border: none;
+	height: 100%;
+	background: var( --color-surface );
+	appearance: none;
+	border-radius: 0;
+	box-sizing: border-box;
+	padding: 0;
+	-webkit-appearance: none;
+
+	&::-webkit-search-cancel-button {
+		-webkit-appearance: none;
+	}
+
+	&:disabled {
+		background: var( --color-surface );
+	}
+
+	&:focus {
+		box-shadow: none;
+		border: none;
+		outline: none;
+
+		&:hover {
+			border: none;
+			box-shadow: none;
+			outline: none;
+		}
+	}
+}
+
+// When search input is opened
+.search.is-open {
+	width: 100%;
+
+	.search__open-icon {
+		color: var( --color-neutral-60 );
+	}
+
+	.search__close-icon {
+		display: inline-block;
+	}
+
+	.search__input,
+	.search__close-icon {
+		opacity: 1;
+	}
+
+	input.search__input[type='search'] {
+		display: block;
+	}
+
+	.search__input-fade {
+		display: flex;
+		align-items: center;
+		border-radius: 0;
+		flex: 1 1 auto;
+		height: 100%;
+		position: relative;
+		font-size: $font-body;
+		&::before {
+			@include long-content-fade(
+				$size: 32px,
+				$z-index: $input-z-index + 2
+			);
+			border-radius: inherit;
+		}
+
+		&.ltr {
+			/*!rtl:ignore*/
+			&::before {
+				@include long-content-fade(
+					$direction: right,
+					$size: 32px,
+					$z-index: $input-z-index + 2
+				);
+				border-radius: inherit;
+			}
+		}
+
+		padding-left: 8px;
+	}
+}
+
+.search.has-open-icon {
+	.search__input-fade {
+		padding-left: 0;
+	}
+}
+
+.search__input-fade .search__text-overlay {
```

Comment:
I wonder if it makes sense to have all the styles nested in `.search`, to ensure we're not bleeding any styles beyond the component.

Also, while on that topic, I wonder if it makes sense to change the wrapper `.search` to a more specific classname. `.search` seems a bit too common, and IMHO there's a big chance to run into conflicts.